instance/tests/models/factories/openedx_appserver.py

```diff
@@ -49,4 +50,68 @@ def make_test_appserver(instance=None, s3=False, server=None):
         appserver.server = server
         appserver.save()
 
+    if status == AppServerStatus.Running:
+        set_appserver_running(appserver)
+    if status == AppServerStatus.ConfigurationFailed:
+        set_appserver_configuration_failed(appserver)
+    elif status == AppServerStatus.Error:
+        set_appserver_errored(appserver)
+    elif status == AppServerStatus.Terminated:
+        set_appserver_terminated(appserver)
+
     return appserver
+
+
+def set_appserver_terminated(appserver):
```

Comment:
Moving these functions from `test_openedx_instance.py` to `factories/openedx_appserver.py` feels bad for many reasons:
they seem to still be test functions only, with no other real use outside tests (we don't want servers moving so fast between states for no reason).
In addition, they are moved into the `factories` folder, but they are not factories; a better place for these functions would be directly somewhere in the server objects: `appserver.set_configuration_failed`.
I'd argue that the existing `make_test_appserver` that we already have at `factories/openedx_appserver.py` isn't at the right place either, because it's used for tests only, and should be somewhere in the tests.
By the way, there's 8-space indentation.
Could you leave these functions near the tests instead of near the implementation?

Reply:
I moved them to the factories module as they help to creating a test instance at a particular state. I also checked that all modules inside `factories` are not being used outside the test context in any way.

> By the way, there's 8-space indentation.
This has been fixed and that what caused the commits difference between this and jazzar/billing-reports-all-commits branch